### DISCUSSION TITLE
Zeffiro downloader and pusher

### DISCUSSION THREAD

**Discussion by sampsapursiainen** [sampsapursiainen/zeffiro_interface#45] [Show and tell]
Link: https://www.youtube.com/watch?v=G2aJZ2fA8ug

This video shows how zeffiro_downloader.m allows setting up a local repository that is automatically updated with the remote origin and how pushing new contents can be done via a pusher tool. The downloader can be  obtained on Matlab's command line via: urlwrite('https://raw.githubusercontent.com/sampsapursiainen/zeffiro_interface/master/zeffiro_downloader.m','zeffiro_downloader.m');

**Comment by SeSodesa** [comment]
I would personally not recommend using the pusher tool. In its current state it is not robust enough to handle merge conflicts and such, and seems to be using the `-A` flag in the `git add ...` phase ([as seen here]), which is generally considered evil by experienced Git users.

[as seen here]: https://github.com/sampsapursiainen/zeffiro_interface/blob/e11a62363d1667e292d5c19196e522b85e9c5e98/plugins/GithubPusher/m/zef_git_push.m#L21

Git should only be used as a separate program, either from the command line or via a mature graphical Git client, such as the ones listed here: <https://git-scm.com/downloads/guis>.

There is a draft of the recommended Git practises for the project in the file <https://github.com/sampsapursiainen/zeffiro_interface/blob/main_development_branch/CONTRIBUTING.md>.
